### PR TITLE
Enhance/#7298 - Navigate to the settings page when clicking "Cancel" when performing ABR setup from the settings page

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #7298 

## Relevant technical choices

* The IB suggests creating a separate test file for `AdBlockingRecoveryApp`. Yet, we've traditionally housed all related tests within the `SetupMain` test file. Given that existing tests already cover some of the behavior of the `cancel` button, new tests have been incorporated into `SetupMain`.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js [hunk 1]
```diff
@@ -186,6 +186,44 @@ describe( 'AdBlockingRecoverySetupCTAWidget - SetupMain', () => {
 				dashboardURL
 			);
 		} );
+
+		it( 'should return to AdSense settings page when `Cancel` button is clicked', async () => {
```
Comment (kuasha420) [inline assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js:190]: This (`should return to AdSense settings page when `Cancel` button is clicked`) and the above test "`should return to dashboard when `Cancel` button is clicked`" sounds the same, we should differentiate between them by adding the perquisite to returning to the specific page in the test name.

File: assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js [hunk 2]
```diff
@@ -186,6 +186,44 @@ describe( 'AdBlockingRecoverySetupCTAWidget - SetupMain', () => {
...
+				value: settingsURL,
+			} );
+
+			const adSenseSettingsURL = `${ settingsURL }#/connected-services/adsense`;
+
+			// eslint-disable-next-line require-await
+			await act( async () => {
+				fireEvent.click( getByRole( 'button', { name: /cancel/i } ) );
+			} );
+
```
Comment (kuasha420) [inline assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js:209]: We should probably duplicate the `mockTrackEvent` assertion as it's a separate but similar test and we'd probably be sure that this was not different. I don't feel super strongly about it, though.

File: assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js [hunk 3]
```diff
@@ -281,6 +319,60 @@ describe( 'AdBlockingRecoverySetupCTAWidget - SetupMain', () => {
 				dashboardURL
 			);
 		} );
+
+		it( 'should return to AdSense settings page when `Cancel` button is clicked', async () => {
```
Comment (kuasha420) [inline assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js:323]: See the above comment about test naming, it applies here as well.

**Review by kuasha420 — CHANGES_REQUESTED** [review]
Thank you @hussain-t, this looks good and works as expected. Left a couple of nitpick-y comments regarding the tests. Please, have a look. 

Cheers.

File: assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js (hunk 2 above)
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js:209]: I'm already asserting the `mockTrackEvent` for the new tests. Please check the code below the `fireEvent.click()`. See: https://github.com/google/site-kit-wp/blob/a282efcd89b3e99e5457325984e180adbc2d06eb/assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js#L210-L214

If this differs from what you intended, could you please clarify it?

File: assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js]: Updated ✅

File: assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js (hunk 3 above)
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js]: Updated ✅

File: assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js (hunk 2 above)
Comment (kuasha420) [reply assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js:209]: Thank you for clarifying, I was under the impression that there was different events for this case, but that is not the case.

**Review by kuasha420 — APPROVED** [review]
Thank you @hussain-t, LGTM :tada:

**Pull request merged**